### PR TITLE
OCPBUGS-21672: http2: trim connetions and buffers

### PR DESCRIPTION
**Pull request by ibihim**
# What

- Replace/bump go.opentelemetry.io/contrib/instrumentation.
- Add capability to disable HTTP/2.
- Add capability to fine-tune resources for HTTP/2.

# Why

- CVE on opentelemetry, which didn't affect KRP, but caused security bots to go haywire.
- CVE-2023-44487: Rapid Reset on Streams enables DDOS attacks.

### CODE REVIEW

File: pkg/filters/auth.go [hunk 1]
```diff
@@ -53,6 +54,20 @@ func WithAuthentication(
...
+		// Do not allow unauthenticated clients to keep these
+		// connections open (i.e. basically degrade them to the
+		// performance of http1 with keep-alive disabled).
+		//
+		// https://github.com/kubernetes/apiserver/blob/master/pkg/endpoints/filters/authentication.go#L107C1-L117C4
+		if req.ProtoMajor == 2 && isAnonymousUser(res.User) {
+			// limit this connection to just this request,
+			// and then send a GOAWAY and tear down the TCP connection
+			// https://github.com/golang/net/commit/97aa3a539ec716117a9d15a4659a911f50d13c3c
+			w.Header().Set("Connection", "close")
```
Comment (s-urbaniak) [inline pkg/filters/auth.go:68]: aren't we supposed to return here? you are continuing the handler chain.

**Comment by s-urbaniak** [comment]
/rename OCPBUGS-21672: http2: trim connetions and buffers

**Comment by xingxingxia** [comment]
Hello all, I pre-merge tested this PR, got panics, details are in OCPBUGS-21672 comment:
```
panic: version string "" doesn't match expected regular expression: "^v(\d+\.\d+\.\d+)"

goroutine 1 [running]:
k8s.io/component-base/metrics.parseVersion({{0x0, 0x0}, {0x0, 0x0}, {0x1bc1454, 0x0}, {0x1bd0b20, 0x28}, {0x0, 0x0}, ...})
	/go/src/github.com/brancz/kube-rbac-proxy/vendor/k8s.io/component-base/metrics/version_parser.go:47 +0x274
k8s.io/component-base/metrics.newKubeRegistry({{0x0, 0x0}, {0x0, 0x0}, {0x1bc1454, 0x0}, {0x1bd0b20, 0x28}, {0x0, 0x0}, ...})
	/go/src/github.com/brancz/kube-rbac-proxy/vendor/k8s.io/component-base/metrics/registry.go:320 +0x119
k8s.io/component-base/metrics.NewKubeRegistry()
	/go/src/github.com/brancz/kube-rbac-proxy/vendor/k8s.io/component-base/metrics/registry.go:335 +0x78
k8s.io/component-base/metrics/legacyregistry.init()
	/go/src/github.com/brancz/kube-rbac-proxy/vendor/k8s.io/component-base/metrics/legacyregistry/registry.go:29 +0x1d
```

**Comment by s-urbaniak** [comment]
(we pair programmed here together)

**Pull request merged**